src/NuGet.Clients/NuGet.CommandLine/Commands/ListCommand.cs

```diff
@@ -37,57 +40,7 @@ public ICollection<string> Source
         [Option(typeof(NuGetCommand), "ListCommandIncludeDelisted")]
         public bool IncludeDelisted { get; set; }
 
-        [SuppressMessage(
-            "Microsoft.Design",
-            "CA1024:UsePropertiesWhereAppropriate",
-            Justification = "This call is expensive")]
-        public IEnumerable<IPackage> GetPackages(IEnumerable<string> listEndpoints)
-        {
-            IPackageRepository packageRepository = GetRepository(listEndpoints);
-            string searchTerm = Arguments != null ? Arguments.FirstOrDefault() : null;
-
-            IQueryable<IPackage> packages = packageRepository.Search(
-                searchTerm,
-                targetFrameworks: Enumerable.Empty<string>(),
-                allowPrereleaseVersions: Prerelease);
-
-            if (AllVersions)
-            {
-                return packages.OrderBy(p => p.Id);
-            }
-            else
-            {
-                if (Prerelease && packageRepository.SupportsPrereleasePackages)
-                {
-                    packages = packages.Where(p => p.IsAbsoluteLatestVersion);
-                }
-                else
-                {
-                    packages = packages.Where(p => p.IsLatestVersion);
-                }
-            }
-
-            var result = packages.OrderBy(p => p.Id)
-                .AsEnumerable();
-
-            // we still need to do client side filtering of delisted & prerelease packages.
-            if (IncludeDelisted == false)
-            {
-                result = result.Where(PackageExtensions.IsListed);
-            }
-            return result.Where(p => Prerelease || p.IsReleaseVersion())
-                       .AsCollapsed();
-        }
-
-        private IPackageRepository GetRepository(IEnumerable<string> listEndpoints)
-        {
-            var repositories = listEndpoints
-                                .Select(RepositoryFactory.CreateRepository)
-                                .ToList();
-
-            var repository = new AggregateRepository(repositories);
-            return repository;
-        }
+        public IListCommandRunner ListCommandRunner { get; set; }
 
         private async Task<IList<KeyValuePair<Configuration.PackageSource, string>>> GetListEndpointsAsync()
```

Comment:
If so, please recursively check for dead code in particular ListCommandResource

---
In reply to: [94630426](https://github.com/NuGet/NuGet.Client/pull/1088#discussion_r94630426) [](ancestors = 94630426)